source/lib/split.ts

```diff
@@ -0,0 +1,28 @@
+type NonEmptyArray<T> = readonly [T, ...(readonly T[])];
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+type NonEmptyString<T extends string> = T extends `${infer _Character}${string}` ? T : never;
+
+type SplitReturnValue<T extends string> = T extends NonEmptyString<T> ? NonEmptyArray<string> : readonly string[];
+
+export function splitByString<Separator extends string>(
+    value: string,
+    separator: Separator
+): SplitReturnValue<Separator> {
+    const builtinSplit = String.prototype.split.bind(value);
```

Comment:
Why do we `bind()` here?

Reply:
I’ve tried to use `call` and `apply`. But this breaks with the TS types. The type definitions of `String.prototype.split` uses overrides to define multiple signatures. Seems like using `call` or `apply` is only able to use the latest override, which expects a splitter object as parameter instead of the separator.